user_sync/config.py

```diff
@@ -312,6 +307,40 @@ def get_rule_options(self):
         if (new_account_type == None):
             new_account_type = user_sync.identity_type.ENTERPRISE_IDENTITY_TYPE
             self.logger.warning("Assuming the identity type for users is: %s", new_account_type)
+
+        after_mapping_hook = None
+        extended_attributes = None
+        extensions_config = self.main_config.get_list_config('extensions', True)
+        if (extensions_config != None):
+            for extension_config in extensions_config.iter_dict_configs():
+                context = extension_config.get_string('context')
+                if context == 'per-user':
+                    if (after_mapping_hook == None):
+                        after_mapping_hook_text = extension_config.get_string('after_mapping_hook')
+                        if (after_mapping_hook_text is not None):
+                            after_mapping_hook = compile(after_mapping_hook_text, '<per-user after-mapping-hook>', 'exec')
+                            extended_attributes = extension_config.get_list('extended_attributes')
+
+                            # [TODO morr 2017-02-27]: Do we really need to pre-create extended dashboard groups here? Or
+                            # could it be done on the fly, when they're encountered in values returned from hook code? If
+                            # the latter, we could do the customer a big favor by not requiring them to be declared in the
+                            # extension config.
+                            #
+                            # This should be revisited once the Complex Mapping feature as a whole is working and has been
+                            # thoroughly tested.
+                            #
+                            for extended_dashboard_group in extension_config.get_list('extended_dashboard_groups'):
+                                group = self.create_dashboard_group(extended_dashboard_group)
```

Comment:
I can see from the code later on that `create_dashboard_group` ignores the group if it already exists as a dashboard group.  This means you are guarding against people specifying groups in their `extended_dashboard_groups` that are already groups in the mapping, which seems like it would be a common (and forgiveable) mistake.  A comment to that effect would be useful here (and also in the docs).